File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    alpidx = [-1] * 26
    
    answer = []
    for i, char in enumerate(s):
        charidx = ord(char) - ord('a')
        
        if alpidx[charidx] == -1:
            answer.append(-1)
        else:
            answer.append(i - alpidx[charidx])
            
        alpidx[charidx] = i
        
    return answer

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

int* solution(const char* s) {
    int slen = strlen(s);
    int* answer = (int*)calloc(slen, sizeof(int));
    if (answer == NULL) return NULL;
    
    int alpidx[26];
    for (int i = 0; i < 26; i++) {
        alpidx[i] = -1;
    }
    
    for (int i = 0; i < slen; i++) {
        int charidx = s[i] - 'a';
        
        if (alpidx[charidx] == -1) {
            answer[i] = -1;
        }
        else {
            answer[i] = i - alpidx[charidx];
        }
        
        alpidx[charidx] = i;
    }
    
    return answer;
}